indra/newview/llviewermessage.cpp

```diff
@@ -2646,19 +2646,21 @@ void process_chat_from_simulator(LLMessageSystem *msg, void **user_data)
         else
         {
             chat.mText = "";
+            std::string chat_text(mesg);
+            auto [msg_without_prefix, is_lua] = LLStringUtil::withoutPrefix(mesg, LUA_PREFIX);
```

Comment:
what about after this `withoutPrefix()` call:
```
std::string prefix;
if (is_lua)
{
    prefix = LUA_PREFIX;
}
```
...
```
case CHAT_TYPE_WHISPER:
    prefix += LLTrans::getString("whisper") + " ";
    break;
case CHAT_TYPE_SHOUT:
    prefix += LLTrans::getString("shout") + " ";
    break;
```
...
```
(line 2673)
chat.mText = prefix + msg_without_prefix;
```
which would make line 2648 moot.